compile_packages.py

```diff
@@ -159,41 +165,129 @@ def build_package(package_path, pack_zip_path, zip_path):
     if os.path.isdir(zip_path):
-        for file in os.listdir(zip_path):
-            print('Deleting', file)
-            os.remove(os.path.join(zip_path, file))
+        if os.listdir(zip_path):
+            raise ValueError("The output directory is not empty")
     else:
         os.makedirs(zip_path, exist_ok=True)
 
-    shutil.rmtree('zips/hammer/', ignore_errors=True)
-
-    path = os.path.join(os.getcwd(), 'packages\\', )
-    
-    # A list of all the package zips.
-    for package in search_folder(zip_path, path):
+    # yield will return found packages one by one, so it is easy to iterate them
+    # without the need to process everything at once
+    for package in search_list_of_dirs(inp_list, zip_path):
         build_package(*package)
+    print('Done!')
 
-    print('Building main zip...')
+    # confusing section, determining whether it need to create main zip and how to name it if not specified
+    # makes it blank by default
+    pack_name = ""
+    if do_zip is None:
+        if conv_bool(input("Zip it all in one file? (y/n) ")):
+            pack_name = 'BEE{}_packages.zip'.format(input('Version: '))
+    else:
+        pack_name = do_zip      # do_zip also can be blank
+    # if pack_name remained blank to this stage, then no zip will be created
+    if pack_name != "":
+        print('Building main zip...')
+        with ZipFile(os.path.join('zips', pack_name), 'w', compression=ZIP_DEFLATED) as zip_file:
+            for file in os.listdir(zip_path):
+                zip_file.write(os.path.join(zip_path, file), os.path.join('packages/', file))
+                print('.', end='', flush=True)
+        print('Done!')
+        if output is not None:
+            print("Moving zip to output destination")
+            shutil.move(os.path.join('zips', pack_name), output)
+            print('Done!')
+            if not args.preserve_temp:
+                print("Deleting temporary files")
+                shutil.rmtree("zips")
+                print("Done!")
+        else:
+            if not args.preserve_temp:
+                print("Deleting temporary files")
+                shutil.rmtree("zips/hammer")
+                if OPTIMISE:
+                    shutil.rmtree("zips/sml")
+                else:
+                    shutil.rmtree("zips/lrg")
+                print("Done!")
+    else:
+        # dont make zip
+        if output is not None:
+            print("Moving files to output destination")
+            if OPTIMISE:
+                shutil.move("zips/sml", output)
+            else:
+                shutil.move("zips/lrg", output)
+            if not args.preserve_temp:
+                print("Deleting temporary files")
+                shutil.rmtree("zips")
+                print("Done!")
+        else:
+            if not args.preserve_temp:
+                print("Deleting temporary files")
+                shutil.rmtree("zips/hammer")
```

Comment:
The `hammer/` folder isn't actually temporary, it's a dump of all the resources, so you could manually load them yourself. I was thinking about releasing that in addition to the packages for Hammer users.

Reply:
oh, thats cool, so maybe i should change text so it says "hammer files" instead of "temporary files"

Reply:
I mean that it shouldn't delete them, maybe move it out of zips then.

Reply:
ok, i updated it a bit, better show what help says, 
![image](https://user-images.githubusercontent.com/86309764/123522611-b9e46d80-d6d7-11eb-91af-ad347cfa853b.png)

Reply:
plus, it won't create any temporary folders, all folders it creates now are output folder and hammer folder, thats all 
only things that are getting deleted by now are unzipped .bee_pack's if an option to zip them all in one zip is chosen